controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/Upgrader.java

```diff
@@ -94,6 +95,7 @@ private void upgrade(ApplicationList applications, Version version) {
         applications = applications.notPullRequest(); // Pull requests are deployed as separate applications to test then deleted; No need to upgrade
         applications = applications.hasProductionDeployment();
         applications = applications.onLowerVersionThan(version);
+        applications = applications.allowMajorVersion(version.getMajor());
```

Comment:
Can we upgrade before the application package has been recompiled on the new major version?